playbooks/deploy-ocp-hybrid-multinode.yml

```diff
@@ -70,6 +77,64 @@
         network_config: "{{ network_config_string | from_yaml }}"
         mac_interface_map: "{{ mac_interface_map_string | from_yaml }}"
 
+# Process optional environment Variables if present
+- name: Mutate worker network configuration if env variables are present
+  hosts: workers
+  gather_facts: false
+  tasks:
+
+    - name: Set env EXTERNAL_INTERFACE variable name
+      ansible.builtin.set_fact:
+        iface_env_var_name: "{{ inventory_hostname + '_EXTERNAL_INTERFACE' }}"
+
+    - name: Set env MAC_ADDRESS variable name
+      ansible.builtin.set_fact:
+        mac_addr_env_var_name: "{{ inventory_hostname + '_MAC_ADDRESS' }}"
+
+    - name: Set fact from environment variable
+      ansible.builtin.set_fact:
+        external_interface_name: "{{ lookup('env', iface_env_var_name) }}"
+        external_interface_mac: "{{ lookup('env', mac_addr_env_var_name) }}"
+
+    - name: Mutate network config if env variables are present
+      when:
+        - external_interface_name is defined
+        - external_interface_name | length > 0
+        - external_interface_mac is defined
+        - external_interface_mac | length > 0
+      ansible.builtin.set_fact:
+        network_config: >-
+          {{
+            network_config | combine({
+              'raw': network_config.raw | combine({
+                'interfaces': network_config.raw.interfaces | map('combine', {
+                  'name': external_interface_name,
+                  'mac-address': external_interface_mac
+                }) | list,
+                'routes': {
+                  'config': network_config.raw.routes.config | map('combine', {
+                    'next-hop-interface': external_interface_name
+                  }) | list
+                }
+              })
+            })
+          }}
```

Comment:
not a PR blocker - but we should convert this to a simpler syntax list J2 or simple yaml 
for readability and easier to debug